python_modules/dagster-test/dagster_test/fixtures/docker_compose.py

```diff
@@ -31,6 +45,27 @@ def docker_compose_cm(
         docker_compose_down(docker_compose_yml, docker_context, service, env_file)
 
 
+def dump_docker_compose_logs(context, docker_compose_yml):
+    global LOGS_DUMPED  # noqa: PLW0603
+    if LOGS_DUMPED:
+        # only do it once
```

Comment:
Under what conditions could it be called multiple times? Are the logs always expected to be the same between subsequent dumps? Can we achieve this without using the global?

Reply:
well don't we call docker_compose_up a bajillion times? they wouldnt' necessarily be the same. i could of course just always dump, i just didn't want to blow up our already ridiculously verbose logs. without a glboal at all? probably not. without an obvious global yes

Reply:
Can't the failure reason be different for each time its called? Or at least for each docker compose file?

Reply:
yup definitely could, this isn't bulletproof